pkg/controller/installation/installation_controller.go

```diff
@@ -86,20 +96,30 @@ type ReconcileInstallation struct {
 	coreClient        *kubernetes.Clientset
 	restConfig        *rest.Config
 	productsToInstall []string
+	context           context.Context
+	cancel            context.CancelFunc
 }
 
 // Reconcile reads that state of the cluster for a Installation object and makes changes based on the state read
 // and what is in the Installation.Spec
 func (r *ReconcileInstallation) Reconcile(request reconcile.Request) (reconcile.Result, error) {
 	instance := &v1alpha1.Installation{}
-	err := r.client.Get(context.TODO(), request.NamespacedName, instance)
+	err := r.client.Get(r.context, request.NamespacedName, instance)
 	if err != nil {
 		if errors.IsNotFound(err) {
 			return reconcile.Result{}, nil
 		}
 		return reconcile.Result{}, err
 	}
 
+	// if the CR is being deleted,
+	// cancel this context to kill all
+	// ongoing requests to the API and exit
+	if instance.DeletionTimestamp != nil {
+		r.cancel() //cancel context
```

Comment:
Could you explain why we need to do this?

Reply:
This sends the cancel to the context which kills any upcoming API requests from any other reconciles that may be in progress.

The issue this is solving is:
--> reconcile starts
--> RHSSO is reconciled
-----> CR is deleted
-----> Cluster clears all items owned by the CR
--> next products are reconciled and recreate all the objects they see are now missing